sql/table.cc

```diff
@@ -5865,6 +5865,7 @@ TABLE_LIST::TABLE_LIST(THD *thd,
                        List<Index_hint> *index_hints_ptr,
                        LEX_STRING *option_ptr)
 {
+  reset();
```

Comment:
This is an inline method that calls bzero, how can we be sure that the compiler won't optimize it away?

Reply:
Thank you @vuvova for sharing your insights, I had the same understanding as you.

It can also be verified with following examples:

Simply moving the memset (or bzero) into the constructor function, from:
https://gcc.godbolt.org/z/6vY5Ma517
```
struct S {
  int i;
  S() {};
};

S* bar() {
  void *buf = malloc(sizeof(S));
  memset(buf, 0, sizeof(S)); 
  S* s = new (buf) S;
  return s;
}
```
to:
https://gcc.godbolt.org/z/aPss1sj59
```
struct S {
  int i;
  S() {  memset(this, 0, sizeof(S)); };
};

S* bar() {
  void *buf = malloc(sizeof(S));
  S* s = new(buf) S;
  return s;
}
```

The assembly code changes from a simple malloc:
```
# Assembly code:
bar():
        mov     edi, 4
        jmp     malloc
```
to malloc + set 0:
```
bar():
        sub     rsp, 8
        mov     edi, 4
        call    malloc
        mov     DWORD PTR [rax], 0
        add     rsp, 8
        ret
```